Content.Server/Supermatter/EntitySystems/SupermatterSystem.cs

```diff
@@ -0,0 +1,391 @@
+using Content.Server.Doors.Components;
+using Content.Shared.Doors;
+using Content.Shared.Damage;
+using Content.Shared.Item;
+using Content.Shared.Whitelist;
+using Content.Shared.Tag;
+using Content.Server.Weapon.Ranged.Ammunition.Components;
+using Content.Shared.Projectiles;
+using Content.Server.Projectiles.Components;
+using Content.Server.Explosion;
+using Content.Server.Radio.EntitySystems;
+using Content.Server.Radio.Components;
+using Content.Server.Chat.Managers;
+using Content.Server.Destructible.Thresholds;
+using Content.Server.Destructible;
+using Content.Shared.Damage.Prototypes;
+using Robust.Shared.GameStates;
+using Robust.Shared.Log;
+using Robust.Shared.Prototypes;
+using System.Collections.Generic;
+using Content.Shared.Acts;
+using Robust.Shared.Analyzers;
+using Robust.Shared.Serialization;
+using Robust.Shared.Serialization.Manager.Attributes;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype;
+using Robust.Shared.Serialization.TypeSerializers.Implementations.Custom.Prototype.List;
+using Robust.Shared.ViewVariables;
+using Content.Shared.Atmos;
+using Content.Server.Alert;
+using Content.Server.Atmos.Components;
+using Content.Shared.Alert;
+using Content.Server.Atmos.EntitySystems;
+using Robust.Shared.Random;
+using Robust.Shared.Timing;
+
+namespace Content.Server.SupermatterSystem{
+    [UsedImplicitly]
+
+    public class SupermatterSystem : EntitySystem
+    {
+        [Dependency] private readonly IPrototypeManager _prototypeManager = default!;
+        [Dependency] private readonly IEntityLookup _lookup = default!;
+        [Dependency] private readonly DamageableSystem _damageable = default!;
+        [Dependency] private readonly AtmosphereSystem _atmosphereSystem = default!;
+        [Dependency] private readonly IMapManager _mapManager = default!;
+        [Dependency] private readonly RadiationPulseSystem _radiationPulseSystem = default!;
+        [Dependency] private readonly RadiationSystem _radiationSystem = default!;
+
+        [Dependency] private IRobustRandom _robustRandom = default!;
+        [Dependency] private IEntityManager _entityManager = default!;
+        private RadioSystem _radioSystem = default!;
+
+        private float _updateAccumulator;
+        private const float _updateTimer = 0.5f;
+        public override void Initialize()
+        {
+            base.Initialize();
+        }
+        public override void Update(float frameTime)
+        {
+            base.Update(frameTime);
+
+            foreach (var supermatter in EntityManager.EntityQuery<SupermatterComponent>())
+            {
+                Update(supermatter, frameTime);
+            }
+        }
+
+        /// <summary>
+        /// Handle outputting radiation based off enery, damage, and gas mix
+        /// </summary>
+        public void HandleRads(SupermatterComponent component, float frameTime)
+        {
+            var Coords = component.Owner.Transform.Coordinates;
```

Comment:
God, just define `ITransformComponent` as a dependency when you make this use resolves and don't do this shit